python/cudf/cudf/core/dataframe.py

```diff
@@ -7362,6 +7343,16 @@ def equals(self, other):
                 return False
         return super().equals(other)
 
+    def _drop_rows_by_labels(
+        self: "cudf.DataFrame", labels: ColumnLike
```

Comment:
If you do:

```
from __future__ import annotations
```

at the top of this file, you can avoid referring to the type of `self` in quotes.